addons/account/models/account.py

```diff
@@ -541,25 +541,25 @@ def _inverse_refund_seq_number_next(self):
                 sequence = journal.refund_sequence_id._get_current_sequence()
                 sequence.number_next = journal.refund_sequence_number_next
 
-    @api.one
     @api.constrains('currency_id', 'default_credit_account_id', 'default_debit_account_id')
     def _check_currency(self):
-        if self.currency_id:
-            if self.default_credit_account_id and not self.default_credit_account_id.currency_id.id == self.currency_id.id:
-                raise ValidationError(_('The currency of the journal should be the same than the default credit account.'))
-            if self.default_debit_account_id and not self.default_debit_account_id.currency_id.id == self.currency_id.id:
-                raise ValidationError(_('The currency of the journal should be the same than the default debit account.'))
+        for rec in self:
```

Comment:
May I ask you to use proper variable/record naming for the iteration?
```journal``` instead of ```rec``` in this method otherwise it is just a big diff in favor of removing without winning a clearer code.

Reply:
@Elkasitu .. @wtaferner got a point

Reply:
but it leads to clearly poor code.

e.g. rec.env[model] - not really an idiom that is used anywhere unless it is specifically required, always self.env. In any case normally they are declared outside of a loop.
Or for rec ins self: rec.x == 'y'. Should just be 
self.write()

or sql queries that are now iterated when a groupby is really whats needed.
or constants that are now initialised multiple times inside a loop.
or functions that clearly require ensure_one() now (like loading a chart of accounts for current company).

sometimes dev is boring.

Reply:
Euhh this is disturbing. You shouldn't refactor over 1.000 lines of code with a macro. This is bound to fail and blow up at some points.. Sometimes a dev needs to do the boring things and take time for these kind of things, although they're not technically challenging. 😉

Reply:
@Yenthe666 don't worry, this was not a blind "apply macro then commit" :wink:

Reply:
@rco-odoo gotcha! So macro and then testing? Then it is good. Thanks for the feedback!